homeassistant/components/motionmount/config_flow.py

```diff
@@ -0,0 +1,173 @@
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Set up the instance."""
+        self.discovery_info: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._show_setup_form()
+
+        info = {}
+        try:
+            info = await self._validate_input(user_input)
+        except (ConnectionError, socket.gaierror):
+            return self.async_abort(reason="cannot_connect")
+        except TimeoutError:
+            return self.async_abort(reason="time_out")
+        except motionmount.NotConnectedError:
+            return self.async_abort(reason="not_connected")
+        except motionmount.MotionMountResponseError:
+            # This is most likely due to missing support for the mac address property
+            # Abort if the handler has config entries already
+            if self._async_current_entries():
+                return self.async_abort(reason="already_configured")
+
+            # Otherwise we try to continue with the generic uid
+            info[CONF_UUID] = config_entries.DEFAULT_DISCOVERY_UNIQUE_ID
+
+        # If the device mac is valid we use it, otherwise we use the default id
+        if info.get(CONF_UUID, EMPTY_MAC) != EMPTY_MAC:
+            unique_id = info[CONF_UUID]
+        else:
+            unique_id = config_entries.DEFAULT_DISCOVERY_UNIQUE_ID
+
+        name = info.get(CONF_NAME, user_input[CONF_HOST])
+
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured(
+            updates={
+                CONF_HOST: user_input[CONF_HOST],
+                CONF_PORT: user_input[CONF_PORT],
+            }
+        )
+
+        return self.async_create_entry(title=name, data=user_input)
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+
+        # Extract information from discovery
+        host = discovery_info.hostname
+        port = discovery_info.port
+        zctype = discovery_info.type
+        name = discovery_info.name.removesuffix(f".{zctype}")
+        unique_id = discovery_info.properties.get("mac")
+
+        self.discovery_info.update(
+            {
+                CONF_HOST: host,
+                CONF_PORT: port,
+                CONF_NAME: name,
+            }
+        )
+
+        if unique_id:
+            # If we already have the unique id, try to set it now
+            # so we can avoid probing the device if its already
+            # configured or ignored
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured(
+                updates={CONF_HOST: host, CONF_PORT: port}
+            )
+
+        self.context.update({"title_placeholders": {"name": name}})
```

Comment:
Abort right before probing if there is already a config entry for the host configured so we don't background probe devices that are already configured

Reply:
`self._async_abort_entries_match({CONF_HOST: host, CONF_PORT: port})`